### ISSUE TITLE
controller.fetch or useFetcher of Resource.detail() are not instanceof Resource

### ISSUE THREAD

**Issue by rtorrente** [coinbase/rest-hooks#1468]
React version (e.g., 16.8.5) : 16.14.0

Concurrent mode no

Rest Hooks version 6.1.7

**Describe the bug**

Result of controller.fetch or useFetcher of Resource.detail() are not instanceof Resource which causes a problem to access calculated properties.

**To Reproduce**

2 codes that are not instance of Resource (Code simplified)

```
const fetch = useFetcher(Car.detail());
const result = fetch({id});
```

```
const {fetch} = useController();
const result = fetch(Car.detail(), {id});
```

For this two requests, result is not a instance of Resource, there is only an object where properties are sent by API Rest. Computed properties defined in Resource class are not present.

**Expected behavior**

For me, the expected behavior is to have a instance of the Resource. It's not explicitly written in the documentation so I could be wrong.

**Additional context**

This issue occurred after trying to replace useStatefullResource which doesn't seem to work (#1467)

**Comment by ntucker** [comment]
You can useCache to look for the resulting denormalized form. This also ensures component re-renders work correctly, etc.

**Comment by andreas-soroko** [comment]
We also ran into this. 

Is there a workaround? Looks like `controller.getResponse` can't be used because the `StateContext` is not public.

**Comment by ntucker** [comment]
@andreas-soroko is it important to keep other entities from state, or just do the same processing to get the classes?

**Comment by ntucker** [comment]
Released in [Rest Hooks 6.5](https://resthooks.io/blog/2022/11/09/Query-getState-SSR-partial-hydration)

**Comment by ntucker** [comment]
Resolution from [useFetch](https://dataclient.io/docs/api/useFetch) and [ctrl.fetch()](https://dataclient.io/docs/api/Controller#fetch) now resolves to a denormalized form (tho does not incorporate any existing entities from state). 

This was first introduced in [rest hooks 8](https://resthooks.io/blog/2023/07/04/v8-release-announcement). Though not all schemas work perfectly until [data client 0.11](https://dataclient.io/blog/2024/04/08/v0.11-queries-querable-usequery)
